src/server.c

```diff
@@ -1070,12 +1070,13 @@ void databasesCron(void) {
         /* Rehash */
         if (server.activerehashing) {
             uint64_t elapsed_us = 0;
+            uint64_t threshold_us = 1 * 1000000 / server.hz / 100;
```

Comment:
This looks like a hard-coded formula and a little hard to see what it means.

Can we use a macro like `INCREMENTAL_REHASHING_THRESHOLD_US_PER_SECOND` to make it clear what this calculated is based on?

```suggestion
            uint64_t threshold_us = INCREMENTAL_REHASHING_US_PER_SECOND / server.hz;
```

with the definition to 10000 to keep the same behaviour

```C
#define INCREMENTAL_REHASHING_US_PER_SECOND 10000
```

Reply:
this hard-coded was taken from active expire:
```
    /* We can use at max 'config_cycle_slow_time_perc' percentage of CPU
     * time per iteration. Since this function gets called with a frequency of
     * server.hz times per second, the following is the max amount of
     * microseconds we can spend in this function. */
    timelimit = config_cycle_slow_time_perc * 1000000 / server.hz / 100;
    timelimit_exit = 0;
    if (timelimit <= 0) timelimit = 1;
```

Reply:
OK, but it is ugly. :)

With this one, I can see that 1000000 is second-to-microsecond and `/ 100` is because the config is in percent.

Reply:
yes, it is a bit ugly, i am going to keep it for now in case we add `active-rehash-cycle` or `active-rehash-effort` something like this to control the CPU, see https://github.com/valkey-io/valkey/pull/977#issuecomment-2323800261